File: src/main/java/com/ExternalCustomerCRM/External/User/Data/Pool/dto/company/response/CompanyDto.java
package com.ExternalCustomerCRM.External.User.Data.Pool.dto.company.response;

import com.ExternalCustomerCRM.External.User.Data.Pool.dto.companynote.response.CompanyNoteDto;
import com.ExternalCustomerCRM.External.User.Data.Pool.dto.contact.response.ContactDto;

import java.util.Set;

public record CompanyDto(
        Long id,
        String name,
        String commercialAgreement,
        CompanyNoteDto companyNoteDto,
        Set<ContactDto> contacts
) {
}

File: src/main/java/com/ExternalCustomerCRM/External/User/Data/Pool/dto/company/CompanyDto.kt
package com.ExternalCustomerCRM.External.User.Data.Pool.dto.company

import com.ExternalCustomerCRM.External.User.Data.Pool.model.CompanyNote
import com.ExternalCustomerCRM.External.User.Data.Pool.model.Contact
import jakarta.persistence.FetchType
import jakarta.persistence.OneToMany
import jakarta.persistence.OneToOne

data class CompanyDto(
    val id : String?,
    val name: String?,
    val commercialAgreement : String?,
    val companyNote : CompanyNote?,
    val contact : Set<Contact>?
){

}